pq-crypto/sike_r2/Makefile

```diff
@@ -16,14 +16,42 @@
 SRCS=fips202.c P434.c
 OBJS=$(SRCS:.c=.o)
 
+# We attempt to use the optimized assembly code. If we can't, we fall
+# back to using the generic C. To use the optimized assembly, we must
+# be on a Linux x86_64 system, using GCC version >= 4.7 (and Clang
+# version >= 3.2, if applicable).
+ifndef S2N_NO_PQ_ASM
+SHOULD_ENABLE_PQ_ASM = 0
 ARCH := $(shell uname -p)
 KERNEL := $(shell uname -s)
 ifeq ($(KERNEL), Linux)
 ifeq ($(ARCH), x86_64)
-ifndef S2N_NO_PQ_ASM
+CC_LENGTH := $(shell expr length "$(CC)")
+ifeq ($(CC_LENGTH), $(shell expr "$(CC)" ":" "g\?cc\-\?[0-9.]*"))
+# Using gcc or cc
+GCC_FULL_VERSION := $(shell $(CC) --version | grep ^$(CC) | sed 's/^$(CC) (.*) //g' | sed 's/ .*//g')
```

Comment:
I think this code is too brittle to include in s2n. I'd rather always attempt to use the PQ ASM and if the compiler is ancient and can't compile the PQ code, then require the user to manually pass in `-DS2N_NO_PQ_ASM` than potentially get it wrong and disable PQ ASM for newer compilers.

Reply:
A more robust attempt would be to attempt compiling the code at configuration time and if it fails disable the feature.

Reply:
I've gone ahead and used the try-compile method for both Make and CMake. I also updated the PQ README with info on how to manually disable the ASM if desired.